src/index.js

```diff
@@ -0,0 +1,32 @@
+
+import updateDOM, {
+  searchForm, checkCity, errorMessage, selectImage,
+} from './modules/dom';
+import parseJSON from './modules/parsejson';
+
+const apiKey = '034b1e6439b4af8304cfb5a8734b6e14';
+searchForm.addEventListener('submit', event => {
+  event.preventDefault();
+
+
+  let data = {};
+  const currentCity = checkCity.value;
+  const metricURL = `https://api.openweathermap.org/data/2.5/weather?q=${currentCity}&appid=${apiKey}&units=metric`;
+  const fehrenhitURL = `https://api.openweathermap.org/data/2.5/weather?q=${currentCity}&appid=${apiKey}&units=imperial`;
+  fetch(metricURL, { mode: 'cors' })
```

Comment:
Very nice use of promises. Notice that one request does not depend on the other, so no need to wait for one to complete before requesting the next one. A good use case for `Promise.all`:

```js
  Promise.all([
    fetch(metricURL).then(response => response.json()),
    fetch(fehrenhitURL).then(response => response.json())
  ]).then(([celsiusData, FahrenheitData]) => {
    updateDOM(parseJSON(celsiusData), parseJSON(FahrenheitData).temp);
  });
```

And you could refactor this a little bit further if you wrap the fetch request on a function, something like this:

```js
  const fetchWeather = async (city, unit) => {
    const response = await fetch(
      `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=${unit}`
    );
    return response.json();
  };

  Promise.all([
    fetchWeather(currentCity, "metric"),
    fetchWeather(currentCity, "imperial")
  ]).then(([celsiusData, FahrenheitData]) => {
    updateDOM(parseJSON(celsiusData), parseJSON(FahrenheitData).tempF);
  });
```

Reply:
Hello @MauricioRobayo, thank you for the feedback, this is my first time to work with promises and I actually did not know anything about Promise.all... I am however confused now because the Fahrenheit when I click on the button to toggle the temperatures, returns 0°F. The metric is working fine but not the imperial. Could you please enlighten me more about this? Thank you.
~~~
const fetchWeather = async (city, unit) => {
    const response = await fetch(
      `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=${unit}`
    );
    return response.json();
  };

  Promise.all([
    fetchWeather(currentCity, "metric"),
    fetchWeather(currentCity, "imperial")
  ]).then(([celsiusData, FahrenheitData]) => {
    updateDOM(parseJSON(celsiusData), parseJSON(FahrenheitData).tempF);
  });
~~~

Reply:
Seems like I made a typo when writing the code, and didn't test it :D, is the property returned by parseJSON called tempF or temp?:

```js
updateDOM(parseJSON(celsiusData), parseJSON(FahrenheitData).tempF);
```

Reply:
And keep the catch after the `then`:

```js
const response = await fetch(
      `https://api.openweathermap.org/data/2.5/weather?q=${city}&appid=${apiKey}&units=${unit}`
    );
    return response.json();
  };

  Promise.all([
    fetchWeather(currentCity, "metric"),
    fetchWeather(currentCity, "imperial"),
  ])
    .then(([celsiusData, FahrenheitData]) => {
      updateDOM(parseJSON(celsiusData), parseJSON(FahrenheitData).temp);
    })
    .catch(() => {
      errorMessage.textContent = "I Couldn't fetch weather for this city 😩.";
    });
```

Reply:
Thank you so much, @MauricioRobayo, It's now working perfect fine!